flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/window/TimeWindow.java

```diff
@@ -220,7 +220,13 @@ public TimeWindowSerializerSnapshot() {
      * @return window start
      */
     public static long getWindowStartWithOffset(long timestamp, long offset, long windowSize) {
-        return timestamp - (timestamp - offset + windowSize) % windowSize;
+        long remainder = (timestamp - offset) % windowSize;
```

Comment:
I know you haven't introduced this method but what do you think about removing this method in favor of the other `TimeWindow#getWindowStartWithOffset`. I am afraid that it is easy to miss that we have the same method twice in the code base.

I am also fine to do this as followup if you think it is too much for this PR.

Reply:
Yes no worries this can also be done outside of the PR.